backend/lib/deal-submitter.js

```diff
@@ -0,0 +1,115 @@
+/** @import {PgPool, Queryable} from '@filecoin-station/deal-observer-db' */
+import Cursor from 'pg-cursor'
+
+/**
+ * Find deals that are eligible to be submitted to a spark api and submits them.
+ *
+ * @param {PgPool} pgPool
+ * @param {string} sparkApiBaseURL
+ * @param {string} dealIngestionAccessToken
+ * @param {number} batchSize
+ * @param {(sparkApiBaseURL: string, dealIngestionAccessToken: string, eligibleDeals: Array) => Promise<void>} submitEligibleDealsFn
+ */
+export const findAndSubmitEligibleDeals = async (pgPool, sparkApiBaseURL, dealIngestionAccessToken, batchSize, submitEligibleDealsFn) => {
```

Comment:
This argument list is quite long for positional arguments.

@bajtos what do you think about also adding this to the linter? Something like positional arguments must be 3 or less?

Reply:
I have made this shorter now by currying `submitDealsToSparkApi` function. Now `findAndSubmitDeals` functions accepts three arguments: pgPool, batchSize, and submitDeals.